File: chapter-1/problems/python/seive_of_erathosthenes.py
def seive_of_erathosthenes(n : int)-> [int]:
    is_prime = [True for i in range(n+1)]
    primes = []
    i = 2

    while i*i<=n:
        if(is_prime[i]):
            j = i*i
            while j<=n:
                is_prime[j] = False
                j+=i
        i+=1
    
    for index in range(len(is_prime)):
        if(is_prime[index]):
            primes.append(index);
    return primes

n = int(input('Enter the value of n : '))
result = seive_of_erathosthenes(n)
print(result)

File: chapter-1/problems/java/seive_of_erathosthenes.java
import java.util.*;

class SieveOfEratosthenes {
    static List<Integer> sieve_of_erathosthenes(int n){
        boolean[] is_prime = new boolean[n+1];
        List<Integer> primes = new ArrayList<>();
        Arrays.fill(is_prime, true);

        for(int i = 2;i*i<=n;i++){
            if(is_prime[i]){
                for(int j = i*i;j<=n;j+=i){
                    is_prime[j] = false;
                }
            }
        }

        for(int i = 2;i<=n;i++){
            if(is_prime[i]){
                primes.add(i);
            }
        }
        return primes;
    }

    static Scanner scanner = new Scanner(System.in);
    public static void main(String[] args){
        System.out.printf("Enter the value of n : ");
        int n = scanner.nextInt();
        List<Integer> primes = sieve_of_erathosthenes(n);
        System.out.printf(primes.toString());
    }
}